books/model-mongodb.js

```diff
@@ -39,12 +40,17 @@ module.exports = function(config) {
 
 
   function getCollection(cb) {
+    if (collection) {
+      setImmediate(function() { cb(null, collection); });
```

Comment:
It depends on which style-guide you follow. Some people expect async methods to always be async because that maintains the run-to-completion semantics and is a cleaner mental model. This is the same argument as to whether promises should be always async (https://promisesaplus.com/#point-34).

I have no strong opinion on this. I had to guess which style you might prefer, and I seemingly guessed wrong :). I wish there was a style guide on this.